packages/@ember/application/type-tests/instance.test.ts

```diff
@@ -19,15 +23,14 @@ expectTypeOf(
   instance.register('service:store-singleton', Store, { singleton: true, instantiate: true })
 ).toEqualTypeOf<void>();
 
-expectTypeOf(instance.lookup<Store>('service:store')).toEqualTypeOf<Store | undefined>();
+expectTypeOf(instance.lookup('service:store')).toEqualTypeOf<unknown>();
```

Comment:
If we add that registry, these will (happily!) flag it up for us as a case where it no longer *equals* that type. :tada: